hs/KernelPerceptron/KernelPerceptronBudgetPython.md

```diff
@@ -0,0 +1,45 @@
+# KernelPerceptronBudgetPython.md
+
+This file provides documentation for the two Python implementations of the Budget Kernel Perceptron.
+
+## Table of Contents
+
+- [Reasoning for a Budget Kernel Perceptron in Python](#reasoning-for-a-budget-kernel-perceptron-in-python)
+- [Differences between Python Implementations](#differences-between-python-implementations)
+- [Required Python Libraries](#required-python-libraries)
+- [Input Dataset Specifications](#Input-dataset-specifications)
+- [Execution](#execution)
+
+## Reasoning for a Budget Kernel Perceptron in Python
+
+The Haskell implementations, extracted from Coq, are the primary implementations used to examine the generalization error and runtime of the Kernel Perceptron variants. However, Haskell is not often used for machine learning applications. Python is one of the primary languages used for machine learning at present due to libraries such as Numpy and Scipy that have made scientific computing more efficient and accessible for students and researchers. The Budget Kernel Perceptron was implemented in Python so that its generalization error and runtime performance could be compared to the Haskell implementation.
+
+## Differences between Python Implementations
```

Comment:
I hadn't seen this file when I made one of the comments above (on "what is the difference between the two Python implementations"). May still be useful to document in the `.py` files, or add a pointer/link to this document.

Reply:
Added comments to both files to point to the documentation